### PR TITLE
Multiple scan XY API Change

### PR DESCRIPTION
**Pull request by CSSFrancis**
This pull request introduces several improvements and bug fixes to the `deapi` client and its test suite. The most significant changes are focused on enhancing the flexibility and robustness of scan pattern handling, improving test reliability and performance, and refining property and acquisition management. Below is a summary of the most important changes, grouped by theme.

**Scan Pattern Handling and Client Improvements:**

* Enhanced `set_xy_array` in `deapi/client.py` to support lists of numpy arrays, handle both 2D and 3D input, ensure integer types, validate shapes, and correctly compute scan dimensions. This improves flexibility and error handling for scan pattern input.
* Modified the property setter to convert boolean values to "On"/"Off" strings, ensuring consistent property value handling.
* Changed the default `pixel_format` in `get_result` to `"AUTO"` for better compatibility.

**Test Suite Reliability and Performance:**

* Updated test fixtures to use `scope="session"` for the `client` fixture, reducing setup/teardown overhead and improving test performance.
* Removed unnecessary `time.sleep` calls and replaced them with more robust idle-waiting logic (e.g., `wait_for_idle(client)`), leading to more reliable and faster tests. [[1]](diffhunk://#diff-7864cfdf61dc0bab36fead911a953abc0502f962e2bb1b4ffd4a265709a80299R19-R33) [[2]](diffhunk://#diff-7864cfdf61dc0bab36fead911a953abc0502f962e2bb1b4ffd4a265709a80299L286-R267) [[3]](diffhunk://#diff-3f981dfa89e7cd373c8a6e89dfac41d8f897c291f8abbcea3383ebfa9edda97fL177)
* Set up a consistent initial state in test fixtures and removed redundant property assignments from individual tests. [[1]](diffhunk://#diff-7864cfdf61dc0bab36fead911a953abc0502f962e2bb1b4ffd4a265709a80299R19-R33) [[2]](diffhunk://#diff-7864cfdf61dc0bab36fead911a953abc0502f962e2bb1b4ffd4a265709a80299L207-L217)

**Test Coverage and Robustness:**

* Added additional property setting and validation in tests, such as explicitly setting hardware binning before testing software binning options, to ensure test correctness. [[1]](diffhunk://#diff-7864cfdf61dc0bab36fead911a953abc0502f962e2bb1b4ffd4a265709a80299R176) [[2]](diffhunk://#diff-7864cfdf61dc0bab36fead911a953abc0502f962e2bb1b4ffd4a265709a80299R186)
* Updated scan and acquisition tests to use more appropriate parameter values and improved error handling for edge cases. [[1]](diffhunk://#diff-7864cfdf61dc0bab36fead911a953abc0502f962e2bb1b4ffd4a265709a80299L314-R316) [[2]](diffhunk://#diff-7864cfdf61dc0bab36fead911a953abc0502f962e2bb1b4ffd4a265709a80299L344-L353)

**File Saving and Loading Tests:**

* Marked slow or unreliable tests with `@pytest.mark.skip(reason="Slow and broken")` to improve test suite reliability and focus on stable tests. [[1]](diffhunk://#diff-46aab8b1d16ab8bdc8a820079f00d9dea8c8031e3d1d54aad9accf34a6289011R32) [[2]](diffhunk://#diff-2db3776e426b320b4ec8fc186cab2812c5264370dde4f1643807aba762cce969R18)
* Moved import of `libertem.api` into the test function to avoid unnecessary imports and potential issues when the library is not available.

### CODE REVIEW

**Comment by CSSFrancis** [comment]
pre-commit.ci autofix

**Comment by CSSFrancis** [comment]
pre-commit.ci autofix

File: deapi/client.py [hunk 1]
```diff
@@ -1246,14 +1267,24 @@ def start_acquisition(
...
+
         if self.width * self.height == 0:
             log.error("  Image size is 0! ")
         else:
             bytesize = 0
             command = self._addSingleCommand(
                 self.START_ACQUISITION,
                 None,
-                [number_of_acquisitions, request_movie_buffer],
+                [number_of_acquisitions, request_movie_buffer] + vb,
```
Comment (CoolFanyu) [inline deapi/client.py:1289]: Only need to add vb if command versions are greater than 15. Otherwise, it will send more information to the old server which does not support this feature

File: deapi/client.py [hunk 2]
```diff
@@ -1919,6 +1993,111 @@ def get_movie_buffer(
...
+            - ``width`` : image width in pixels
+            - ``height`` : image height in pixels
+            - ``data_type`` : :class:`~deapi.DataType` of each pixel
+        """
+        command = self._addSingleCommand(self.GET_VIRTUAL_IMAGE_INFO, None, None)
+        response = self._sendCommand(command)
+
+        info = VirtualImageInfo()
+        if response:
+            values = self.__getParameters(response.acknowledge[0])
```
[the remark this answers is not in the document]
Comment (CoolFanyu) [reply deapi/client.py:2024]: No need to consider the commandVer here since it's a new function

**Review by CoolFanyu — COMMENTED** [review]
At start_acquisition, need to consider back compatible.

**Comment by CSSFrancis** [comment]
pre-commit.ci autofix
